Many tests under LayoutTests/pointer-lock can run on iOS

aprotyas opened the pull request:
#### c71f9ab03a693473b25e64dfaddc8bcb0e76605f
<pre>
Many tests under LayoutTests/pointer-lock can run on iOS
<a href="https://bugs.webkit.org/show_bug.cgi?id=297354">https://bugs.webkit.org/show_bug.cgi?id=297354</a>
<a href="https://rdar.apple.com/158246804">rdar://158246804</a>

Reviewed by Lily Spiniolas and Aditya Keerthi.

Most of the requisite test harness support is in place and a sizeable
feature subset is testable, so let us start by enabling various layout
tests under the pointer-lock directory.

We mock pointing device availability in all tests, and as a drive-by
fix, migrate away from the outdated js-test-[pre|post] idiom.

* LayoutTests/platform/ios/TestExpectations:
* LayoutTests/platform/mac/TestExpectations:
Drive-by: pointer-lock/lock-lost-on-alert.html is no longer failing on
macOS.
* LayoutTests/pointer-lock/lock-already-locked.html:
* LayoutTests/pointer-lock/lock-element-not-in-dom.html:
* LayoutTests/pointer-lock/lock-lost-on-alert.html:
* LayoutTests/pointer-lock/lock-lost-on-navigation.html:
* LayoutTests/pointer-lock/locked-element-removed-from-dom.html:
* LayoutTests/pointer-lock/mouse-event-api.html:
* LayoutTests/pointer-lock/pending-locked-element-removed-from-dom.html:
* LayoutTests/pointer-lock/pointer-lock-api.html:
* LayoutTests/pointer-lock/pointerlock-interface.html:

Canonical link: <a href="https://commits.webkit.org/298666@main">https://commits.webkit.org/298666@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/33ec92f21fb68b829d67b052e93c5d3715ab0593

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/116178 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/35839 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/26384 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/122235 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/66737 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/33cfba7c-a7b8-4011-9673-963f47f96960) 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/36533 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/44427 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/88267 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/42800 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/7d6ac930-889c-4988-a985-cbe39047a817) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/119127 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/29143 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/104253 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/68678 "Passed tests") | | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/28258 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/22357 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/65916 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/98534 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/22494 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/125384 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/43072 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/32342 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/96993 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/43437 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/100438 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/96777 "Passed tests") | | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/42051 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/19948 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/39019 "Built successfully") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/18569 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/42959 "Built successfully") | | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/42426 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/45761 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/44130 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/33ec92f21fb68b829d67b052e93c5d3715ab0593. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/49357

aprotyas commented:
Thank you for the reviews!

aprotyas committed:
Many tests under LayoutTests/pointer-lock can run on iOS
https://bugs.webkit.org/show_bug.cgi?id=297354
rdar://158246804

Reviewed by Lily Spiniolas and Aditya Keerthi.

Most of the requisite test harness support is in place and a sizeable
feature subset is testable, so let us start by enabling various layout
tests under the pointer-lock directory.

We mock pointing device availability in all tests, and as a drive-by
fix, migrate away from the outdated js-test-[pre|post] idiom.

* LayoutTests/platform/ios/TestExpectations:
* LayoutTests/platform/mac/TestExpectations:
Drive-by: pointer-lock/lock-lost-on-alert.html is no longer failing on
macOS.
* LayoutTests/pointer-lock/lock-already-locked.html:
* LayoutTests/pointer-lock/lock-element-not-in-dom.html:
* LayoutTests/pointer-lock/lock-lost-on-alert.html:
* LayoutTests/pointer-lock/lock-lost-on-navigation.html:
* LayoutTests/pointer-lock/locked-element-removed-from-dom.html:
* LayoutTests/pointer-lock/mouse-event-api.html:
* LayoutTests/pointer-lock/pending-locked-element-removed-from-dom.html:
* LayoutTests/pointer-lock/pointer-lock-api.html:
* LayoutTests/pointer-lock/pointerlock-interface.html:

Canonical link: https://commits.webkit.org/298666@main
